core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/FilterRule.scala

```diff
@@ -44,7 +43,15 @@ class FilterRule(val settings: Settings)
         case Left(_: Unresolvable) =>
           Rejected()
         case Right(filter) =>
-          Fulfilled(blockContext.withAddedContextHeader(Header(Name.transientFilter, filter)))
+          BlockContextUpdater[B] match {
```

Comment:
IMO we should firstly check the block context type and then process only the ones which makes sense in context of the rule

Reply:
we can also refactor `isAllowedForDLS` - it's rather related to search block context that request

Reply:
please write a test where `isAllowedForDLS` returns false (current impl is a bit weird for me)